xtask/src/ci.rs

```diff
@@ -0,0 +1,89 @@
+use std::path::PathBuf;
+
+use xshell::pushd;
+
+use crate::{cmd, Result};
+
+const MSRV: &str = "1.45";
+
+/// Task to run CI tests.
+pub struct CiTask {
+    /// Which version of Rust to test against.
+    version: Option<String>,
+
+    /// The root of the workspace.
+    project_root: PathBuf,
+}
+
+impl CiTask {
+    pub(crate) fn new(version: Option<String>, project_root: PathBuf) -> Self {
+        Self { version, project_root }
+    }
+
+    pub(crate) fn run(self) -> Result<()> {
+        match self.version.as_deref() {
+            Some("msrv") => self.build_msrv(),
+            Some("stable") => self.build_stable(),
+            Some("nightly") => self.build_nightly(),
+            Some(_) => Err("Wrong Rust version specified.".into()),
+            None => {
+                self.build_msrv()?;
+                self.build_stable()?;
+                self.build_nightly()
+            }
+        }
+    }
+
+    fn build_msrv(&self) -> Result<()> {
+        {
+            let _p = pushd(self.project_root.join("ruma"))?;
+            cmd!("rustup run {MSRV} cargo build --features ruma-events,ruma-api,ruma-appservice-api,ruma-client-api,ruma-federation-api,ruma-identity-service-api,ruma-push-gateway-api --quiet").run()?;
+        }
+        {
+            let _p = pushd(self.project_root.join("ruma-client"))?;
```

Comment:
You previously introduced a macro so the pushd thing is less verbose. Mind introducing that again since there's so many of these calls here? Maybe not as `cmd!` but as `cmd_in!` / `cmd_wd!` (for working directory) or something like that.